### PR TITLE
strings, arrays and safe-box challenge

### PR DESCRIPTION
**Pull request by laia89**
strings, arrays and safe-box challenge

### CODE REVIEW

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+            alert('Sorry, we have temporary problem, try again later.');
+        });
+
+
+        
+});
+var resultsArtists = new ResultsList();
+
+//Create List of artists by artist
+var DEFAULT_IMAGE = 'https://i.pinimg.com/originals/37/2a/2d/372a2d5e8a32991bb19982271d0762fe.jpg';
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: Wrong default image (there shouldn't be a default image)

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+resultsArtists.onItemClick(function (id) {
+    logic.retrieveAlbumsByArtistId(id)
+    .then(function (albums) {
+debugger;
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: This shouldn't be here: "debugger"

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+var resultsTracks = new ResultsList();
+
+
+//List detail of track
+resultsTracks.onItemClick(function (id) {
+
+    logic.retrieveTrackById(id)
+        .then(function (track) {
+            debugger;
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: remove "debugger"

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                id: track.id,
+                text: track.name
+            };
+        }));
+
+        detailContainer.clear();
+
+    })
+    .catch(function (error) {
+        alert('Sorry, we have temporary problem, try again later.');
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: Show error code

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        .then(function (track) {
+            debugger;
+            //var detail = new DetailPanel(beer.name, beer.style.description, beer.labels ? beer.labels.medium : DEFAULT_IMAGE);
+            var detail = new DetailPanel(track.name, track.style.description, track.labels ? track.labels.medium : DEFAULT_IMAGE);
+
+            detailContainer.clear();
+            detailContainer.appendChild(detail.element);
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: Show error code

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+var detailContainer = document.createElement('div');
+
+detailContainer.clear = function () {
+    this.innerHTML = '';
+};
+
+function updateActiveAlbumById(id) {
+    var albumsContainer = document.getElementById('albumsContainer');
+    var prevActiveAlbum = albumsContainer.querySelectorAll('ul > li[data-active="true"]')[0];
+    prevActiveAlbum.setAttribute('ative', 'false');
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: ative ?

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+    var newActiveAlbum = document.querySelectorAll('albumsContainer');
+    newActiveAlbum.setAttribute('ative', 'true');
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/scripts/main.js]: ative ?

**Review by robert-barcelona — COMMENTED** [review]
Very good but some errors!

File: staff/laia-chorro/spotify-app/index.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (SergioCastilloSorribes) [inline staff/laia-chorro/spotify-app/index.html]: Títol!!!!

File: staff/laia-chorro/spotify-app/scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,128 @@
+// my presentation logic
```
Comment (SergioCastilloSorribes) [inline staff/laia-chorro/spotify-app/scripts/main.js]: Falta Use Strict!!!!

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 2 above)
Comment (SergioCastilloSorribes) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Debbuger in code!!!!!!!!!!???? Oh my GOD!!!!

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 1 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Artists default image removed from next version!
Thanks.

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 2 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Sorry! Now there are not more debuggers in the code.

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 3 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Again, no more debuggers now.

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.2.0/scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
+'use strict'
+
+var search = new SearchNavBar(),
```
Comment (robert-barcelona) [inline staff/laia-chorro/spotify-app/.versions/spotify-app-1.2.0/scripts/main.js:3]: Piensa en cambiar 'var' a 'const' y 'let'

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 4 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Fixed in 1.2.0 version :)

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 5 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: Fixed in 1.2.0 version :)

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/index.html [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (SergioCastilloSorribes) [inline staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/index.html:7]: 'Document' is the worst semantic title i had ever seen!

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/scripts/main.js [hunk 12]
```diff
@@ -0,0 +1,126 @@
+// my presentation logic
```
Comment (SergioCastilloSorribes) [inline staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/scripts/main.js:1]: Missed Use strict!!!!

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.1.0/index.html [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (SergioCastilloSorribes) [inline staff/laia-chorro/spotify-app/.versions/spotify-app-1.1.0/index.html:7]: Again!!! You may change the title!!!

File: staff/laia-chorro/spotify-app/scripts/main.js (hunk 6 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/scripts/main.js]: I didn't need this class after all, so I removed the 'ative/active' attr in 1.2.0 version

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/scripts/main.js (hunk 12 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/.versions/spotify-app-1.0.0/scripts/main.js:1]: 'use strict' added!

File: staff/laia-chorro/spotify-app/.versions/spotify-app-1.2.0/scripts/main.js (hunk 10 above)
Comment (laia89) [reply staff/laia-chorro/spotify-app/.versions/spotify-app-1.2.0/scripts/main.js:3]: this version is done in ECMA5

**Pull request merged**